### PR TITLE
RangeMap function in new Analyzer class, optimized regions list creation (only created once now), tests.

### PR DESCRIPTION
**Pull request by sophbohr22**

### CODE REVIEW

File: smaps/src/test/java/com/google/smaps/FileParserTest.java [hunk 1]
```diff
@@ -34,28 +34,27 @@
...
 
   @Test
   public void fileNotFound() {
     // Tests that an empty list is successfully returned with nonexistent file.
-    String nonexistentFilePathname = "../fake-file.txt";
-    List<Region> list = FileParser.getRegionList(nonexistentFilePathname);
-    assertEquals(0, list.size());
+    Analyzer.makeRegionList("../fake-file.txt");
+    List<Region> list = Analyzer.getRegionList();
+    assertEquals(null, list);
```
Comment (maggieroemer) [inline smaps/src/test/java/com/google/smaps/FileParserTest.java]: Is there an assertNull helper?

**Review by maggieroemer — APPROVED** [review]
LGTM, nice work on that Range Map Sophie!

File: smaps/src/test/java/com/google/smaps/FileParserTest.java (hunk 1 above)
Comment (sophbohr22) [reply smaps/src/test/java/com/google/smaps/FileParserTest.java]: Yes there is! I'll use that instead.

File: smaps/src/main/java/com/google/smaps/Analyzer.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  static void makeRangeMap(List<Region> regionsList) {
+    // Create the range map.
+    addressRangeMap = TreeRangeMap.create();
+    // Go through every region and add it to the range map.
+    for (int i = 0; i < regionsList.size(); i++) {
+      Region curR = regionsList.get(i);
+      // Parse the addresses with base 16 because they are hexadecimal.
+      BigInteger start = new BigInteger(curR.startLoc(), 16);
+      BigInteger end = new BigInteger(curR.endLoc(), 16);
+      // Put the region into the range map with the address range (inclusive, exclusive).
```
Comment (segfault-the-real-segfault) [inline smaps/src/main/java/com/google/smaps/Analyzer.java]: I suggest [inclusive, exclusive) to reflect the mathematical meaning of [ and )

**Review by segfault-the-real-segfault — APPROVED** [review]

**Pull request merged**